tests/api/test_sources_endpoint.py

```diff
@@ -337,3 +338,30 @@ def test_timeout_url(self, client, source_add_user, mock_icon_urls):
         )
 
         assert resp.data[0]['detail'] == 'Could not download/process image.'
+
+    def test_canonical_source(self, client, source_add_change_user, mock_icon_urls):
+        test_data = get_post_body(canonical=True)
+        resp = client.post(
+            self.endpoint,
+            json.dumps(test_data),
+            content_type='application/vnd.api+json',
+            HTTP_AUTHORIZATION=source_add_change_user.authorization(),
+        )
+        assert resp.status_code == 201
+        data_one = flatten_write_response(resp)
+
+        assert data_one['source']['canonical']
+
+        source_url = '{}{}/'.format(self.endpoint, data_one['source']['id'])
+        test_two_data = get_post_body(id=data_one['source']['id'], canonical=False)
```

Comment:
So here you're just updating the same source to be not canonical? Could you maybe add a comment or something to make this more clear?